src/main/java/edu/studyup/serviceImpl/EventServiceImpl.java

```diff
@@ -29,28 +29,32 @@ public Event updateEventName(int eventID, String name) throws StudyUpException {
 		return event;
 	}
 
+	//BUG #2: Inefficient use of keySet iterator
+	//SOLUTION: use entrySet iterator
 	@Override
 	public List<Event> getActiveEvents() {
 		Map<Integer, Event> eventData = DataStorage.eventData;
 		List<Event> activeEvents = new ArrayList<>();
 
-		for (Integer key : eventData.keySet()) {
-			Event ithEvent = eventData.get(key);
+		
+		for (Map.Entry<Integer, Event> key : eventData.entrySet()) {
```

Comment:
I totally agree with you that we should use the entrySet() iterator since it is more efficient than the previous one.

Reply:
It's a matter of efficiency, for certain. We wouldn't want to be iterating over the same objects using the keySet() iterator.